gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/service/JavaServiceDistributionPlugin.java

```diff
@@ -158,9 +165,13 @@ public void apply(Project project) {
                     task.getAddJava8GcLogging().set(distributionExtension.getAddJava8GcLogging());
                     task.getJavaHome().set(distributionExtension.getJavaHome());
                     task.getEnv().set(distributionExtension.getEnv());
-                    task.setClasspath(
-                            jarTask.get().getOutputs().getFiles().plus(
-                                    distributionExtension.getProductDependenciesConfig()));
+                    if (distributionExtension.getEnableManifestClasspath().get()) {
```

Comment:
This is eager and so will not respect user configuration. I'd suggest centralizing the logic to switch on `enableManifestClassPath` into a single `afterEvaluate` block so that its lazy and easier to read